common/src/main/java/com/microsoft/identity/common/internal/dto/PrimaryRefreshTokenRecord.java

```diff
@@ -0,0 +1,252 @@
+
+import static com.microsoft.identity.common.internal.dto.PrimaryRefreshTokenRecord.SerializedNames.*;
+
+public class PrimaryRefreshTokenRecord extends Credential {
+
+    public static class SerializedNames extends Credential.SerializedNames {
+        /**
+         * String of family id.
+         */
+        public static final String FAMILY_ID = "family_id";
+
+        /**
+         * String of session_key.
+         */
+        public static final String SESSION_KEY = "session_key";
+
+        /**
+         * String of session_key_rolling_date.
+         */
+        public static final String SESSION_KEY_ROLLING_DATE = "session_key_rolling_date";
+
+        /**
+         * String of prt_protocol_version.
+         */
+        public static final String PRT_PROTOCOL_VERSION = "prt_protocol_version";
+    }
+
+    /**
+     * 1st Party Application Family ID.
+     */
+    @SerializedName(FAMILY_ID)
+    private String mFamilyId;
+
+    /**
+     * PRT expiry time. This value is returned from the server as refresh_token_expires_in that
+     * should be added to local time.
+     */
+    @SerializedName(EXPIRES_ON)
+    private String mExpiresOn;
+
+    /**
+     * Session key of PRT.
+     */
+    @SerializedName(SESSION_KEY)
+    private String mSessionKey;
+
+    /**
+     * The version of the PRT Protocol being used, version 3 supports no device_id.
+     */
+    @SerializedName(PRT_PROTOCOL_VERSION)
+    private String mPrtProtocolVersion;
+
+    /**
+     * Session key expiry time. This value is determined by the client and set to 60 days after the
+     * session key was initally issued
+     */
+    @SerializedName(SESSION_KEY_ROLLING_DATE)
+    private String mSessionKeyRollingDate;
+
+    /**
+     * Gets the family_id.
+     *
+     * @return The family_id to get.
+     */
+    public String getFamilyId() {
+        return mFamilyId;
+    }
+
+    /**
+     * Sets the family_id.
+     *
+     * @param familyId The family_id to set.
+     */
+    public void setFamilyId(final String familyId) {
+        mFamilyId = familyId;
+    }
+
+    public boolean isExpired(final String expires) {
+        // Init a Calendar for the current time/date
+        final Calendar calendar = Calendar.getInstance();
+        final Date validity = calendar.getTime();
+        // Init a Date for the accessToken's expiry
+        long epoch = Long.valueOf(expires);
+        final Date expiresOn = new Date(
+                TimeUnit.SECONDS.toMillis(epoch)
+        );
+        return expiresOn.before(validity);
+    }
+
+    @Override
+    public boolean isExpired() {
+        return isExpired(getExpiresOn());
+    }
+
+    /**
+     * Gets the expires_on.
+     * @return The expires_on to get.
+     */
+    public String getExpiresOn() {
+        return mExpiresOn;
+    }
+
+    /**
+     * Sets the expires_on.
+     * @param expiresOn The expires_on to set.
+     */
+    public void setExpiresOn(String expiresOn) {
```

Comment:
Can we put the units in the parameter name, since I think this is epoch seconds.

Reply:
It's same as [AccessTokenRecord](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/040cbd91ccbfa46a72ab6584efe95ee1bfe27363/common/src/main/java/com/microsoft/identity/common/internal/dto/AccessTokenRecord.java#L309-L311), we can just leave it?

Reply:
Eh.  I'd like to change it everywhere, if I could.  Since it's a String, it doesn't tell the developer anything about what's in it - for all a programmer knows without looking at the implementation it's an ISO date string.  Not a huge issue, just something that could be better.